tests/test_GitRepository.cc

```diff
@@ -0,0 +1,283 @@
+        {
+            if (gul14::starts_with(elm.path_name, "unit_test_1") || gul14::starts_with(elm.path_name, "unit_test_2"))
+            {
+                REQUIRE(elm.handling == "unchanged");
+                REQUIRE(elm.changes == "unchanged");
+            }
+        }
+
+        REQUIRE(gl.get_last_commit_message() == "Add second sequence");
+    }
+
+
+    /**
+     * Change two files, but only stage one of them.
+     * 1) Manipulate both files of unit_test_1
+     * 2) check if status of them is modified, but unstaged
+     * 3) stage file1 of unit_test_1
+     * 4) file1 should be staged and file0 still be unstaged
+    */
+    SECTION("Add by path")
+    {
+        GitRepository gl{"sequences"};
+
+        create_testfiles("unit_test_1", 2, "Add by path");
+
+        std::vector<FileStatus> stats = gl.status();
+        REQUIRE(stats.size() != 0);
+        for(auto elm: stats)
+        {
+            if (gul14::starts_with(elm.path_name, "unit_test_1/file"))
+            {
+                REQUIRE(elm.handling == "unstaged");
+                REQUIRE(elm.changes == "modified");
+            }
+            else if (gul14::starts_with(elm.path_name, "unit_test_2/file"))
+            {
+                REQUIRE(elm.handling == "unchanged");
+                REQUIRE(elm.changes == "unchanged");
+            }
+        }
+
+        auto ret = gl.add_files({"unit_test_1/file1.txt"});
+
+        // No errors should have occur 
+        REQUIRE(ret.size() == 0);
+
+        stats = gl.status();
+        REQUIRE(stats.size() != 0);
+        for(auto elm: stats)
+        {
+            if (gul14::starts_with(elm.path_name, "unit_test_1/file0"))
+            {
+                REQUIRE(elm.handling == "unstaged");
+                REQUIRE(elm.changes == "modified");
+            }
+            else if (gul14::starts_with(elm.path_name, "unit_test_1/file1"))
+            {
+                REQUIRE(elm.handling == "staged");
+                REQUIRE(elm.changes == "modified");
+            }
+            else if (gul14::starts_with(elm.path_name, "unit_test_2/file"))
+            {
+                REQUIRE(elm.handling == "unchanged");
+                REQUIRE(elm.changes == "unchanged");
+            }
+        }
+    }
+
+    /**
+     * remove a directory and check if the repository status notices
+     * 1) Delete unit_test_2
+     * 2) Files should be automatically staged for deletion
+     * 3) Commit removal
+     * 4) check if files are gone from status
+     * 5) check if files are gone from filesystem
+     */
+    SECTION("Delete Directory")
+    {
+        // Create Git Library
+        GitRepository gl{"sequences"};
+        
+        std::filesystem::path mypath = "unit_test_2";
+
+        gl.remove_directory(mypath);
+
+        std::vector<FileStatus> stats = gl.status();
+
+        // every file in unit_test_2 should have the tag deleted
+        REQUIRE(stats.size() != 0);
+        for(auto elm: stats)
```

Comment:
This copies each element, which is wasteful. Better use const references:
```suggestion
        for (const auto& elm : stats)
```